src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -449,4 +468,40 @@ private void checkCursorAny(Cursor c, CursorKind... kinds) {
         }
         throw new IllegalArgumentException("Invalid cursor kind");
     }
+
+    private <D extends Declaration> D withDeclarationString(D decl, Cursor cursor) {
+        String declString = switch (decl) {
+            case Declaration.Constant _ -> null; // do nothing for enum constants
+            case Typedef _ -> declarationString(cursor, true); // always expand typedefs
+            default -> {
+                // heuristic, try w/o expanding first, and check if there are <anonymous> strings
+                String cursorString = declarationString(cursor, false);
+                if (cursorString.contains("\\(unnamed (struct|union|enum) at")) {
+                    // the output contains anonymous definitions, fallback and expand them
+                    cursorString = declarationString(cursor, true);
+                }
+                yield cursorString;
```

Comment:
So, if I understand directly, this is different from what is described in the PR body. e.g. we will try to expand struct types as just  `struct Foo`, unless we also find anonymous decls, in which case we expand to `struct Foo { int x; int y; }` ?

Reply:
yes, this change was described in a follow up comment